### PR TITLE
Exclude windows labeled nodes from running cpu frequency getter

### PR DESCRIPTION
**Pull request by irfanurrehman**
Implements https://vmturbo.atlassian.net/browse/OM-69572

The details of implementation are in the jira issue.
As a gist:
After this PR, the windows nodes will be automatically identified using the below default labels (nodes labelled with either):
`kubernetes.io/os: windows`
`beta.kubernetes.io/os: windows`

This can be overridden with a strict set of labels passed as kubeturbo cmd line argument `cpufreq-job-exclude-node-labels`.
eg. `--cpufreq-job-exclude-node-labels=kubernetes.io/os=windows,product=turbo`

Tested both the default and the cmd line argument to be working as expected. 
Logs for a windows node (first discover, default cpu freq hasn't been filled from linux nodes):
```
E0419 03:19:05.953938       1 kubelet_client.go:322] Failed to get machine[ip-192-168-18-158.ec2.internal] cpu.frequency from kubelet: "https://192.168.18.158:10250/spec" was not found. Will try pod based getter.
W0419 03:19:05.953984       1 kubelet_client.go:326] Node ip-192-168-18-158.ec2.internal seems to be a windows node, default cpu frequency will be used.
I0419 03:19:05.953995       1 kubelet_monitor.go:255] node-ip-192-168-18-158.ec2.internal cpuFrequency = 2000.00MHz
```

Logs for a windows node (further discoveries, default cpu freq has been filled from linux nodes):
```
E0419 03:29:03.197224       1 kubelet_client.go:322] Failed to get machine[ip-192-168-18-158.ec2.internal] cpu.frequency from kubelet: "https://192.168.18.158:10250/spec" was not found. Will try pod based getter.
W0419 03:29:03.197260       1 kubelet_client.go:326] Node ip-192-168-18-158.ec2.internal seems to be a windows node, default cpu frequency will be used.
I0419 03:29:03.197269       1 kubelet_monitor.go:255] node-ip-192-168-18-158.ec2.internal cpuFrequency = 2528.97MHz
```

### CODE REVIEW

**Review by chlam4 — APPROVED** [review]

File: cmd/kubeturbo/app/kubeturbo_builder.go [hunk 1]
```diff
@@ -328,7 +335,11 @@ func (s *VMTServer) Run() {
 	// Collect target and probe info such as master host, server version, probe container image, etc
 	k8sTAPSpec.CollectK8sTargetAndProbeInfo(kubeConfig, kubeClient)
 
-	kubeletClient := s.CreateKubeletClientOrDie(kubeConfig, kubeClient, s.BusyboxImage, s.UseNodeProxyEndpoint)
+	excludeLabelsMap, err := mapFromSelector(s.CpufreqJobExcludeNodeLabels)
+	if err != nil {
+		glog.Fatalf("Invalid cpu frequency exclude node labels: %v", err)
```
[the remark this answers is not in the document]
Comment (irfanurrehman) [reply cmd/kubeturbo/app/kubeturbo_builder.go]: Updated!

File: cmd/kubeturbo/app/kubeturbo_builder.go [hunk 2]
```diff
@@ -186,6 +191,7 @@ func (s *VMTServer) AddFlags(fs *pflag.FlagSet) {
 	fs.StringSliceVar(&s.sccSupport, "scc-support", defaultSccSupport, "The SCC list allowed for executing pod actions, e.g., --scc-support=restricted,anyuid or --scc-support=* to allow all.")
 	fs.StringVar(&s.ClusterAPINamespace, "cluster-api-namespace", "default", "The Cluster API namespace.")
 	fs.StringVar(&s.BusyboxImage, "busybox-image", "busybox", "The complete image uri used for fallback node cpu frequency getter job.")
+	fs.StringVar(&s.CpufreqJobExcludeNodeLabels, "cpufreq-job-exclude-node-labels", "", "The comma saperated list of key=value node label pairs to be excluded from running job based cpufrequency getter.")
```
[the remark this answers is not in the document]
Comment (irfanurrehman) [reply cmd/kubeturbo/app/kubeturbo_builder.go]: Updated!

File: pkg/discovery/util/node_util_test.go [hunk 3]
```diff
@@ -0,0 +1,86 @@
+package util
+
+import (
+	"testing"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestNodeMatchesLabels(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (irfanurrehman) [reply pkg/discovery/util/node_util_test.go:10]: Updated!

**Comment by irfanurrehman** [comment]
Thanks @chlam4 for the review. Updated as per your suggestions. PTAL!

**Review by enlinxu — APPROVED** [review]

**Pull request merged**